django/db/models/base.py

```diff
@@ -834,6 +834,11 @@ def save_base(self, raw=False, force_insert=False,
         # Once saved, this is no longer a to-be-added instance.
         self._state.adding = False
 
+        # Readonly fields are computed by the Database. If there are any,
+        # we should fetch their value. (refresh_from_db does nothing if the
+        # field list is empty)
+        self.refresh_from_db(fields=[field.name for field in self._meta.readonly_fields])
```

Comment:
Hmm, a whole object seems overkill because it would mean one does not get the same kind of object when calling create in a MySQL and PGSQL environment, but maybe we could just mark those fields as deferred, so there would be a query when one tries to access the fields (maybe with a way to load them all at once ?)